src/node.js

```diff
@@ -390,6 +391,78 @@
     }
   };
 
+  startup.processPromises = function() {
+    // this should be kept in sync with PromiseRejectEvent in v8.h
+    var kPromiseRejectWithNoHandler = 0;
+    var kPromiseHandlerAddedAfterReject = 1;
+
+    var timers = NativeModule.require('timers');
+    var unhandledPromises = new WeakMap();
+    // A hack to work around uniterable WeakMaps.
+    var unhandledPromiseKeys = [];
+    var fullTurnQueued = false;
+
+    function queueFullTurn() {
+      if (!fullTurnQueued) {
+        fullTurnQueued = true;
+        timers.setImmediate(function() {
+          process.nextTick(afterFullTurn);
```

Comment:
this causes weird behaviour: https://github.com/iojs/io.js/pull/758#issuecomment-73428399